hw3/condition.py

```diff
@@ -0,0 +1,75 @@
+"""
+Environment:
+    system: Ubuntu 16.04.4, x86_64
+    Python: 3.5.2
+
+Task: output numbers from 0..100 in order.
+First thread outputs even numbers.
+Second thread outputs odd numbers.
+
+Type of synchronization object: Condition.
+"""
+from threading import Condition
+from threading import Thread
+from time import sleep
+
+condition = Condition()
+
+result = []
+
+SLEEP_TIME = 0.01  # seconds
+TOTAL_RANGE = 101
+
+
+def even_print():
+    """Prints even values. Acts as a producer."""
+
+    for value in range(TOTAL_RANGE):
+
+        if value % 2 == 0:
+            sleep(SLEEP_TIME)
+            condition.acquire()
```

Comment:
it would be better to use `with condition:`
```
with condition:
            print("'{function_name}': {value}".
                  format(function_name=even_print.__name__, value=value))
            result.append(value)
            condition.notify()
```